.github/workflows/build-gem.yml

```diff
@@ -0,0 +1,165 @@
+name: Build gem
+
+on:
+  workflow_dispatch:
+    inputs:
+      push:
+        description: Push gem
+        required: true
+        type: boolean
+        default: true
+  push:
+    branches:
+      - '**'
```

Comment:
Is it necessary to build the gem on every push on any branch? Wonder what's would be your thoughts on this?